File: src/main/java/org/chousik/collection/builder/collectors/FloatCollector.java
package org.chousik.collection.builder.collectors;

import org.chousik.collection.validators.IValidator;
import org.chousik.exception.ScriptExecutionError;

import java.util.function.Function;

public class FloatCollector extends NumberCollector<Float>{

    @Override
    public Float ask(String name, IValidator<Float> validator) throws ScriptExecutionError {
        return askNumber(name, validator, Float::parseFloat);
    }
}

File: src/main/kotlin/collection/builder/collectors/FloatCollector.kt
package org.chousik.collection.builder.collectors

import org.chousik.collection.validators.IValidator

class FloatCollector : NumberCollector<Float>() {
    override fun ask(name: String, validator: IValidator<Float?>): Float {
        return askNumber(name, validator, java.lang.Float::parseFloat)
    }
    }